syntax = "proto3";

option java_package = "com.tad.submission.grpc";

service SubmissionTaskService {
  rpc HandleSubmissionTask (SubmissionTaskRequest) returns (SubmissionTaskResponse);
  rpc HandleCreateSubmission (CreateSubmissionRequest) returns (CreateSubmissionResponse);
}

message SubmissionTaskRequest {
  string task_type = 1; // "UPLOAD_LINK", "UPDATE_RESULT"
  string data = 2;
}

message SubmissionTaskResponse {
  bool success = 1;
}

message CreateSubmissionRequest {
  string user_id = 1;
  string laboratory_id = 2;
}

message CreateSubmissionResponse {
  string submission_id = 1;
}